pkg/healthcheck/healthcheck_test.go

```diff
@@ -1774,6 +1781,215 @@ data:
+			tlsSecretScheme:  string(corev1.SecretTypeTLS),
+			schemeInConfig:   string(corev1.SecretTypeTLS),
+			expectedErr:      nil,
+		},
+
+		{
+			checkDescription: "fails when config scheme is linkerd.io/tls but secret scheme is kubernetes.io/tls in config is different than the one in the issuer secret",
+			tlsSecretScheme:  string(corev1.SecretTypeTLS),
+			schemeInConfig:   k8s.IdentityIssuerSchemeLinkerd,
+			expectedErr:      errors.New("key crt.pem containing the issuer certificate needs to exist in secret linkerd-identity-issuer if --identity-external-issuer=false"),
+		},
+
+		{
+			checkDescription: "fails when config scheme is kubernetes.io/tls but secret scheme is linkerd.io/tls in config is different than the one in the issuer secret",
+			tlsSecretScheme:  k8s.IdentityIssuerSchemeLinkerd,
+			schemeInConfig:   string(corev1.SecretTypeTLS),
+			expectedErr:      errors.New("key ca.crt containing the trust anchors needs to exist in secret linkerd-identity-issuer if --identity-external-issuer=true"),
+		},
+
+		{
+			checkDescription:   "fails when cert dns is wrong",
+			certificateDNSName: "wrong.linkerd.cluster.local",
+			expectedErr:        errors.New("invalid credentials: x509: certificate is valid for wrong.linkerd.cluster.local, not identity.linkerd.cluster.local"),
+			fileNameToSave:     "wrong-domain",
+		},
+
+		{
+			checkDescription: "fails when cert is not valid yet",
+			lifespan: &lifeSpan{
+				starts: time.Date(2100, 1, 1, 1, 1, 1, 1, time.UTC),
+				ends:   time.Date(2101, 1, 1, 1, 1, 1, 1, time.UTC),
+			},
+			expectedErr:    errors.New("certificate not valid before: 2100-01-01T01:00:51Z"),
+			fileNameToSave: "not-valid-yet",
+		},
+
+		{
+			checkDescription: "fails when cert is expired",
+			lifespan: &lifeSpan{
+				starts: time.Date(1989, 1, 1, 1, 1, 1, 1, time.UTC),
+				ends:   time.Date(1990, 1, 1, 1, 1, 1, 1, time.UTC),
+			},
+			expectedErr:    errors.New("certificate not valid anymore. Expired at: 1990-01-01T01:01:11Z"),
+			fileNameToSave: "expired",
+		},
+
+		{
+			checkDescription: "warns if certificate expires to soon",
+			lifespan: &lifeSpan{
+				starts: time.Date(1989, 1, 1, 1, 1, 1, 1, time.Local),
+				ends:   time.Now().AddDate(0, 0, 1),
+			},
+			expectedWarning: errors.New("certificate is expiring sooner than 60 days from now"),
+		},
+	}
+
+	for id, testCase := range testCases {
+		testCase := testCase
+
+		if testCase.certificateDNSName == "" {
+			testCase.certificateDNSName = "identity.linkerd.cluster.local"
+		}
+		if testCase.lifespan == nil {
+			testCase.lifespan = &lifeSpan{
+				starts: time.Now().AddDate(-1, 0, 0),
+				ends:   time.Now().AddDate(1, 0, 0),
+			}
+		}
+
+		if testCase.schemeInConfig == "" && testCase.tlsSecretScheme == "" {
+			testCase.schemeInConfig = k8s.IdentityIssuerSchemeLinkerd
+			testCase.tlsSecretScheme = k8s.IdentityIssuerSchemeLinkerd
+		}
+
+		t.Run(fmt.Sprintf("%d", id), func(t *testing.T) {
```

Comment:
should the test name use `testCase.checkDescription`?

Reply:
Was trying to be consistent with [this](https://github.com/linkerd/linkerd2/blob/master/pkg/healthcheck/healthcheck_test.go#L1757), although it does not make sense to me that the description is used only for code documentation purposes. Should we change it for all tests in this file?

Reply:
seems weird to me.  I'd prefer to have the description in there but :shrug: not a big deal